enhancements/sig-policy/59-ansible-everyevent-mode/README.md

```diff
@@ -0,0 +1,209 @@
+to a policy violation. It is meant to track the clusters **currently** in the state based on the `eventHook` value in
+the `PolicyAutomation` object and the appropriate automation was run for those clusters. Such a history can be found on
+the Ansible Tower/Ansible Automation Platform job history.
+
+Several minutes later, the `enable-etcd-encryption` policy becomes noncompliant on `cluster2`. The Governance Policy
+Propagator controller would run the Ansible job from the Hub while passing the `extra_var` of
+`target_clusters: ["cluster2"]`. It would then set the following status on the `PolicyAutomation` object.
+
+```yaml
+status:
+  clustersWithEvent:
+    cluster1:
+      automationStartTime: "2022-06-22T23:58:30Z"
+    cluster2:
+      automationStartTime: "2022-06-22T23:59:30Z"
+```
+
+Several minutes later, the `enable-etcd-encryption` policy becomes compliant on `cluster1`. The Governance Policy
+Propagator controller would not run an Ansible job and set the following status on the `PolicyAutomation` object.
+
+```yaml
+status:
+  clustersWithEvent:
+    cluster2:
+      automationStartTime: "2022-06-22T23:59:30Z"
+```
+
+Now consider the case where the user made a change to their Ansible job and wants the Governance Policy Propagator
+controller to rerun the Ansible jobs for all current noncompliant managed clusters. The user would set the existing
+annotation of `policy.open-cluster-management.io/rerun` to `true` on the `PolicyAutomation` object and thus signal the
+Governance Policy Propagator controller to ignore `status.clustersWithEvent` and run an Ansible job for every
+noncompliant managed cluster.
+
+### Limiting the Number of Ansible Jobs
+
+If a user has a policy that they are concerned with often going between compliant and non-compliant, they may not want
+to run an Ansible job every time the cluster becomes noncompliant in order to not overwhelm Ansible Tower/AAP. In this
+case, a new optional spec field of `delayAfterRunSeconds` (follows the
+[API naming conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions))
+would be added so the user could specify the minimum amount of seconds before an automation can be retriggered on the
+same cluster. This would default to 0 seconds.
+
+If a user sets `delayAfterRunSeconds` to `600` for example, and the cluster becomes noncompliant a second time and stays
+noncompliant past the 600 seconds, the Ansible job would be run again for that cluster. When the `delayAfterRunSeconds`
+value is set, the `PolicyAutomation` controller would not remove the `status.clustersWithEvent` entry for the cluster
+until it has become compliant and the number of seconds set in `delayAfterRunSeconds` have elapsed since the
+`automationStartTime` value. From an implementation point of view, this would mean requeing the reconcile request to
+that point in time.
```

Comment:
The addition of `eventTime` should let it work as described, so that is good.

I still think there's the potential here that AutomationJobs will be created on a loop. The frequency of that loop is limited by the delay here, but I think the alternate behavior I described could prevent that loop.

The question is: if the policy returns to noncompliance before the delay is up, should the automation just be run again? I think there is a case to made for "no" - in this case the automation didn't really work, and it might be bad to just re-run it because that creates more objects in k8s.

I'd like to see it clarified in the goals/non-goals of the proposal. As you now describe it, I think the goal is to limit the rate that AnsibleJobs are created, with a non-goal of preventing loops from creating an infinite number of Jobs.

Reply:
@JustinKuli right, this behavior is as intended because the Ansible job likely won't resolve the policy violation but merely just trigger some external action such as creating a service now ticket or creating some kind of alert, which we would want to run repeatedly. So the goal is to just reduce the amount of Ansible jobs that get created for an unstable policy.

I added this in the goal and non-goal section as you suggested.